quotaclimat/data_processing/mediatree/channel_program.py

```diff
@@ -0,0 +1,63 @@
+import modin.pandas as pd
+import logging
+import os
+
+def get_programs():
+    current_dir = os.path.dirname(os.path.abspath(__file__))
+    json_file_path = os.path.join(current_dir, 'channel_program.json')
+    df_programs = pd.read_json(json_file_path)
+   
+    df_programs['start'] = pd.to_datetime(df_programs['start'], format='%H:%M')
+    df_programs['end'] = pd.to_datetime(df_programs['end'], format='%H:%M')
+
+    logging.info(f"df_programs : {df_programs.head(2).to_string()}")
+
+    return df_programs
+
+
+
+def add_channel_program(df: pd.DataFrame): 
+    logging.debug("Adding channel program")
+   
+    try:
+        df_programs = get_programs()
+        merged_df = merge_program_subtitle(df, df_programs)
+        return merged_df
+    except (Exception) as error:
+        logging.error("Could not merge program and subtitle df", error)
+        raise Exception
+
+def merge_program_subtitle(df_subtitle: pd.DataFrame, df_program: pd.DataFrame):
+    merged_data = []
+    for index, subtitle in df_subtitle.iterrows():
+        start_time = pd.to_datetime(subtitle['start'].strftime("%H:%M"), format="%H:%M")
+        logging.debug(f"start_time {start_time}")
+        # with Monday=0 and Sunday=6.
+        start_weekday = int(subtitle['start'].dayofweek)
+        logging.debug(f"start_weekday {start_weekday}")
+
+        matching_rows = df_program[
```

Comment:
main logic here, we merge 2 different dataframe :
1. first dataframe first are tv/audio data
2. second dataframe is program info

If the tv/audio data is the same weekday, same channel_name, and between start and end time of a program, we apply 2 values : program name and program type -(arcom type)